wallet-wasm/src/lib.rs

```diff
@@ -1254,3 +1255,71 @@ pub extern "C" fn decrypt_with_password(
         -1
     }
 }
+
+#[no_mangle]
+pub extern "C" fn redemption_private_to_address(
+    private_ptr: *const c_uchar,
+    protocol_magic: u32,
+    out: *mut c_uchar,
+) -> u32 {
+    let priv_key = unsafe {
+        let slice: &[u8] = std::slice::from_raw_parts(private_ptr, redeem::PRIVATEKEY_SIZE);
+        redeem::PrivateKey::from_slice(slice).unwrap()
+    };
+    let pub_key = priv_key.public();
+    let addr_type = address::AddrType::ATRedeem;
+    let addr_data = address::SpendingData::RedeemASD(pub_key.clone());
+    let magic = cardano::config::ProtocolMagic::new(protocol_magic);
+    let addr_attr = address::Attributes::new_bootstrap_era(None,magic.into());
+    let address = address::ExtendedAddr::new(addr_type, addr_data, addr_attr));
+    let address_bytes = cbor!(address).unwrap();
+    unsafe { write_data(&address_bytes, out) }
+    return address_bytes.len() as u32;
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct WalletRedeemInput {
+    protocol_magic: cardano::config::ProtocolMagic,
+    redemption_key: redeem::PrivateKey,
+    input: TxIn,
+    output: TxOut,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct WalletRedeemOutput {
+    cbor_encoded_tx: Vec<u8>,
+}
+
+#[no_mangle]
+pub extern "C" fn xwallet_redeem(
+    input_ptr: *const c_uchar,
+    input_sz: usize,
+    output_ptr: *mut c_uchar,
+) -> i32 {
+    let data: WalletRedeemInput = input_json!(output_ptr, input_ptr, input_sz);
+    let &mut txbuilder = txbuild::TxBuilder::new();
+    txbuilder.add_input(&data.input.convert(), data.output.value.0);
+    txbuilder.add_output_value(&data.output.convert());
+    let tx: tx::Tx = jrpc_try!(
+        output_ptr,
+        txbuilder.make_tx()
+            .map_err(|e| JsValue::from_str(&format! {"{:?}", e}))
+    );
+    let witness = tx::TxInWitness::redeem(
+        data.protocol_magic, &data.redemption_key, &tx.0.id());
```

Comment:
Here I've imagined how new functionality for redemption signing might look:
![image](https://user-images.githubusercontent.com/5585355/51396333-f1867d00-1b4f-11e9-97e1-4ac83c873d4d.png)

But of course will update to actual new functionality when @NicolasDP adds `signRedeem` to new API